### PR TITLE
Fixes for new foreign key constraints 

### PR DESCRIPTION
**Pull request by Abyss777**
Solution to store `long == 0` as `NULL` in DB is not very nice, but simple. The other one is create "Zero-Group" but I do not like it, to many unpredictable side effects and changes.

 I tested a bit, but did not see any regressions in other places we store `long`s.

Also there was bug in SQL request and warning in browser when editing groups

PS: also I can replace `if (value == 0)` to `if (value == 0 && name.toLowerCase().equals("groupid"))`. It will minimize side effects (if they exist)

### CODE REVIEW

File: src/org/traccar/database/QueryBuilder.java [hunk 1]
```diff
@@ -175,7 +175,11 @@ public QueryBuilder setInteger(String name, int value) throws SQLException {
     public QueryBuilder setLong(String name, long value) throws SQLException {
         for (int i : indexes(name)) {
             try {
-                statement.setLong(i, value);
+                if (value == 0) {
+                    statement.setNull(i, Types.INTEGER);
+                } else {
+                	statement.setLong(i, value);
```
Comment (tananaev) [inline src/org/traccar/database/QueryBuilder.java]: Replace tab with spaces.

**Comment by tananaev** [comment]
My only concern is that if we really want to save 0 for something, it won't work. I guess if we don't have anything else of type `long`, we should be fine.

**Comment by Abyss777** [comment]
We have. `server.zoom` and `user.zoom` other `long` are `id`s that can't be `0`.
Maybe it is really better to use `if (value == 0 && name.toLowerCase().equals("groupid"))`, rather ugly but without side effects.

**Comment by tananaev** [comment]
I think we should add another method `setLong` with third parameter. Something like `nullIfZero`. Then in `setObject` check if method name ends with `Id` (case sensitive), then call method with `true`, otherwise call old version.

**Comment by Abyss777** [comment]
Done

File: src/org/traccar/database/QueryBuilder.java [hunk 2]
```diff
@@ -185,6 +185,23 @@ public QueryBuilder setLong(String name, long value) throws SQLException {
         return this;
     }
 
+    public QueryBuilder setLong(String name, long value, boolean nullIfZero) throws SQLException {
```
Comment (tananaev) [inline src/org/traccar/database/QueryBuilder.java]: Please make `setLong` function with 2 parameters call this one to avoid duplication.

**Comment by Abyss777** [comment]
OK

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**